app.py

```diff
@@ -8,157 +10,216 @@
-        f.close()
-
-        if not result:#  != 'true':
-            #if error != 'incorrect-captcha-sol':
-            app.logger.info("ReCaptcha fail: %r, %r", oform, d)
-            #g.recaptcha_args += "&error=" + error
+        response = requests.post(
+            "https://www.google.com/recaptcha/api/siteverify", payload
+        )
+        # result = f.readline().replace("\n","")
+        # error = f.readline().replace("\n","")
+        response_text = json.loads(response.text)
+        result = response_text["success"]
+
+        if not result:  #  != 'true':
+            # if error != 'incorrect-captcha-sol':
+            app.logger.info("ReCaptcha fail: %r, %r", payload, result)
+            # g.recaptcha_args += "&error=" + error
             return False
 
     except:
-        #g.recaptcha_args += "&error=unknown"
+        # g.recaptcha_args += "&error=unknown"
         return False
     return True
 
-@app.route('/haxx', methods=["POST"])
+
+@app.route("/haxx", methods=["POST"])
 def haxx():
-    OUI_LIST = [i.decode('hex') for i in open(os.path.join(app.root_path, 'oui_list.txt')).read().split("\n") if len(i)==6]
-    g.recaptcha_args = 'k=%s' % app.config['RECAPTCHA_PUBLICKEY']
+
+    OUI_LIST = [
+        binascii.unhexlify(i)
+        for i in open(os.path.join(app.root_path, "oui_list.txt")).read().split("\n")
+        if len(i) == 6
+    ]
+    g.recaptcha_args = "k=%s" % app.config["RECAPTCHA_PUBLICKEY"]
     dt = datetime.utcnow() - timedelta(1)
-    delta = (dt - datetime(2000, 1, 1))
+    delta = dt - datetime(2000, 1, 1)
     timestamp = delta.days * 86400 + delta.seconds
 
     try:
-        mac = ''.join([chr(int(request.form[i],16)) for i in "abcdef"])
-        template = TEMPLATES[request.form['region']]
-        bundle = 'bundle' in request.form
+        mac = "".join([chr(int(request.form[i], 16)) for i in "abcdef"])
+        template = TEMPLATES[request.form["region"]]
+        bundle = "bundle" in request.form
     except:
         return _index("Invalid input.")
     if not captcha_check():
         return _index("Are you a human?")
 
     if mac == "\x00\x17\xab\x99\x99\x99":
-        app.logger.info('Derp MAC %s at %d ver %s bundle %r', mac.encode('hex'), timestamp, request.form['region'], bundle)
+        app.logger.info(
+            "Derp MAC %s at %d ver %s bundle %r",
+            mac.encode("hex"),
+            timestamp,
+            request.form["region"],
+            bundle,
+        )
         return _index("If you're using Dolphin, try File->Open instead ;-).")
 
-    if not any([mac.startswith(i) for i in OUI_LIST]):
-        app.logger.info('Bad MAC %s at %d ver %s bundle %r', mac.encode('hex'), timestamp, request.form['region'], bundle)
-        return _index("The exploit will only work if you enter your Wii's MAC address.")
-
-
-    key = hashlib.sha1(mac+"\x75\x79\x79").digest()
-
-    blob = bytearray(open(os.path.join(app.root_path, template),'rb').read())
+    if not any:
```

Comment:
Something went terribly wrong here, this block doesn't make any sense. "not any" is always false.